indico/modules/events/models/events.py

```diff
@@ -1070,9 +1073,17 @@ def is_user_speaker(self, user):
                 .has_rows())
 
     @contextmanager
-    def force_event_locale(self):
-        # TODO: once we have an event locale, force that one
-        with force_locale(None):
+    def force_event_locale(self, user=None):
+        locale = default_locale = self.default_language
+        if not user:
+            user = session.user
+        if user:
+            user_settings = user.settings
+            locale = user_locale = user_settings.get('lang')
+            if not user_settings.get('force_language') and user_locale not in self.supported_languages:
+                locale = default_locale
```

Comment:
I'd like to have a unit test for this. Especially since the function right now feels a bit convoluted (nope, I didn't look into simplifying it yet), having it covered with tests would be appreciated.

Reply:
Hi @ThiefMaster, I prepared a unit test but it isn't working well, locally I have issues debugging the pytest in v3, 
`pytest_redis.executor.UnixSocketTooLong: Unix Socket path is longer than 104 allowed on your system: `

Reply:
Full traceback please; and a `print()` somewhere to see the path it tries to use there would be useful.

Reply:
```
E           pytest_redis.executor.UnixSocketTooLong: Unix Socket path is longer than 104 allowed on your system: /private/var/folders/jk/kkn6jl017pxgdmkb6zjt06100000gn/T/pytest-of-vasantvohra/pytest-6/pytest-redis-redis_proc0/redis.29696.sock. It's probably due to the temporary directory configuration. You can configure that for python by changing TMPDIR envvar, add for example `--basetemp=/tmp/pytest` to your pytest command or add `addopts = --basetemp=/tmp/pytest` to your pytest configuration file.
```

Reply:
I was able to fix pytest, by adding `--basetemp=/tmp/pytest` as mentioned in traceback.
`addopts = -rsfEw --cov indico --cov-report html --no-cov-on-fail --basetemp=/tmp/pytest`

Reply:
Yes, I think the problem was your weird tmp path which is way too long for unix sockets... usually it's `/tmp/pytest-....` ;)